web/source/text/domKbdInterface.ts

```diff
@@ -101,20 +101,37 @@ namespace com.keyman.text {
   //The following entry points are defined but should not normally be used in a keyboard, as OSK display is no longer determined by the keyboard
   KeyboardInterface.prototype['hideHelp'] = function(): void {
     let keyman = com.keyman.singleton;
-    keyman.osk._Hide(true);
+    keyman.osk.startHide(true);
   }
 
   KeyboardInterface.prototype['showHelp'] = function(Px: number, Py: number): void {
     let keyman = com.keyman.singleton;
-    keyman.osk._Show(Px,Py);
+    if(keyman.osk instanceof osk.FloatingOSKView) {
+      keyman.osk.presentAtPosition(Px,Py);
+    } else {
+      keyman.osk.present();
+    }
   }
 
   KeyboardInterface.prototype['showPinnedHelp'] = function(): void {
     let keyman = com.keyman.singleton;
     if(keyman.osk instanceof osk.FloatingOSKView) {
-      keyman.osk.userPositioned=true;
+      // An old KMW bug previously auto-unset the affected field when this function was
+      // used by CJK keyboards during rule processing.  As a result, we need to condition
+      // on whether or not:
+      // 1.  The active keyboard is CJK
+      // 2.  A keyboard rule is actively processing.
+      //
+      // If BOTH are true, we do NOT mutate keyman.osk.userPositioned.
+      // Otherwise, not all conditions are met, so we still allow OSK pinning.
+      if(!keyman.core.activeKeyboard.isCJK || !this.ruleBehavior) {
+        keyman.osk.userPositioned=true;
+      }
```

Comment:
The "old KMW bug":  the old `_Show` method always called `.loadCookie()` as part of its operation.  If the OSK were previously unpinned, that call would immediately unset `.userPositioned` before any actual pinning-related logic ran - which prevented OSK pinning.  Which, from an end-user perspective with CJK keyboards, seemed fine.

Because the `osk.present()` method no longer calls `.loadCookie`, removing that call caused the unwanted CJK pinning bug to surface.  `Keyboard_japanese` contains a hardcoded call to `showPinnedHelp` _which should not pin the OSK_.  Fortunately, the two conditions mentioned in the comment are sufficient to block the unwanted pinning effect without the need to modify the keyboard.

This sound fine, @mcdurdin?

Reply:
I think that sounds fine. The APIs used by CJK were always a little rough and really need some deep TLC but for now just keeping them working in the same way as before, as far as possible, should be okay.